src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -131,76 +129,75 @@ public ExpenseOD editExpenseInMonth(ExpenseID expenseIDEdit) throws EntityExcept
         MonthlyStorage month = session.getMonthlyData(expenseID.getMonthID());
         ArrayList<Expense> monthExpenseList = month.getExpenseData();
         ArrayList<Category> categoryList = month.getCategoryData();
-        Expense selectedExpense;
 
         try {
             Category selectedCategory = findCategory(categoryList, expenseID.getOldCategory());
-
-            if(Objects.equals(expenseIDEdit.getOldCategory(), "Other")){
-                selectedExpense = findExpense(recurringExpenseList, expenseID.getOldExpense());
-            }else{selectedExpense = findExpense(monthExpenseList, expenseID.getOldExpense());}
+            Expense selectedExpense = findExpense(monthExpenseList, expenseID.getOldExpense());
 
             double valueDouble = toDouble(expenseIDEdit.getValue());
             if (valueDouble < 0) {
-                //4. Expense budget less than 0 fail
-                ExpenseOD expenseODFailEdit = new ExpenseOD("Expense budget can't be less than $0. Please try again!");
+                //Expense value less than 0 fail: When a user tries to edit the expense value with a negative number.
+                ExpenseOD expenseODFailEdit = new ExpenseOD("Expense value can't be less than $0. Please try again!");
                 return expenseOB.fail(expenseODFailEdit);
-            }//&&!expenseID.getIsRecurringExpense()
+            }
             if(!Objects.equals(expenseIDEdit.getName(), expenseIDEdit.getOldExpense())){
-                if(checkHaveSameNameInList(monthExpenseList)) {
-                    //1. Repeated name fail
+                if(checkHaveSameNameInList(monthExpenseList, expenseID.getName())) {
+                    // Repeated name fail: When a user tries to edit the expense name to a existing expense in month.
                     ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
                     return expenseOB.fail(expenseODFailEdit);
                 }}
-            if (changeInRecurringInfo()) {// 4. Same recurring expense name fail
```

Comment:
I actually think you could keep this line, it helps tell you whether or not there is a change, making other code redundant (like the name check in the recurring list can be removed because we already know the expense was not recurring before).